integration_tests/src/main/python/window_function_test.py

```diff
@@ -609,6 +610,34 @@ def do_it(spark):
     assert_gpu_and_cpu_are_equal_collect(do_it, conf={'spark.rapids.sql.hasNans': 'false'})
 
 
+@pytest.mark.skipif(is_before_spark_320(), reason="Only in Spark 3.2.0 is IGNORE NULLS supported for lead and lag by Spark")
+@allow_non_gpu('WindowExec', 'Alias', 'WindowExpression', 'Lead', 'Lag', 'Literal', 'WindowSpecDefinition', 'SpecifiedWindowFrame')
+@ignore_order(local=True)
+@pytest.mark.parametrize('d_gen', all_basic_gens, ids=meta_idfn('agg:'))
+@pytest.mark.parametrize('c_gen', [LongRangeGen()], ids=meta_idfn('orderBy:'))
+@pytest.mark.parametrize('b_gen', [long_gen], ids=meta_idfn('orderBy:'))
+@pytest.mark.parametrize('a_gen', [long_gen], ids=meta_idfn('partBy:'))
+def test_window_aggs_lead_lag_ignore_nulls_fallback(a_gen, b_gen, c_gen, d_gen):
+    data_gen = [
+            ('a', RepeatSeqGen(a_gen, length=20)),
+            ('b', b_gen),
+            ('c', c_gen),
+            ('d', d_gen),
+            ('d_default', d_gen)]
+
+    assert_gpu_sql_fallback_collect(
+        lambda spark: gen_df(spark, data_gen),
+        'Lead',
+        "window_agg_table",
+        '''
+        SELECT
+            LEAD(d, 5) IGNORE NULLS OVER (PARTITION by a ORDER BY b,c) lead_d_5,
+            LEAD(d, 2, d_default) IGNORE NULLS OVER (PARTITION by a ORDER BY b,c) lead_d_2_default,
+            LAG(d, 5) IGNORE NULLS OVER (PARTITION by a ORDER BY b,c) lag_d_5,
```

Comment:
Should there be separate queriess for LEAD vs. LAG?  By putting both expressions in the same query with the same partitioning and ordering, don't we risk the test passing just because one of the expressions gets the logic right and kicks the operation off the GPU but the other expression tagging logic could still be incorrect?

Reply:
Yup I was being lazy...